src/state_point.F90

```diff
@@ -223,7 +240,28 @@ subroutine write_state_point()
             filter_group = create_group(tally_group, "filter " // &
                  trim(to_str(j)))
 
-            call write_dataset(filter_group, "type", tally%filters(j)%type)
+            ! Write name of type
+            select case (tally%filters(j)%type)
```

Comment:
I'm not sure if it would simplify the code elsewhere, but it seems that the use of a hash table / dictionary to store the filter enumerated types mapped to string types would be cleaner than a ``case`` statement which will inevitably grow larger in the future.